prov/efa/src/rdm/rxr_env.c

```diff
@@ -281,39 +247,56 @@ void rxr_env_initialize()
 /**
  * @brief Fetch the value of the environment variable FI_EFA_USE_DEVICE_RDMA
  *
- * This function fetches the change in the value of FI_EFA_USE_DEVICE_RDMA
- * if any, after it was defined and set in rxr_env_define_use_device_rdma() 
- * 
+ * This function fetches the value of FI_EFA_USE_DEVICE_RDMA from the
+ * user's environment.
+ *
+ * Environments with non-boolean (1, true, yes, on) values for the variable
+ * will print to stderr and abort the program.
+ *
+ * If environment variable is not set, return RDMA device support as
+ * is the default setting for the variable.
+ *
  * @return		0 - If FI_EFA_USE_DEVICE_RDMA is set to 0/false/no/off
- * 			1 - If FI_EFA_USE_DEVICE_RDMA is set to 1/true/yes/on
- * 			If FI_EFA_USE_DEVICE_RDMA is not set, then the value
- * 			returned from rxr_env_get_default_use_device_rdma()
- * 			will be the return value of this function.
+ *			1 - If FI_EFA_USE_DEVICE_RDMA is set to 1/true/yes/on
  */
-int rxr_env_get_use_device_rdma()
+bool rxr_env_get_use_device_rdma()
 {
-	int ret, use_device_rdma;
+	int ret;
+	int param_val;
+	bool hw_support;
 
-	/* Get the default value of using device's RDMA capability */
-	use_device_rdma = rxr_env_get_default_use_device_rdma();
+	hw_support = efa_device_support_rdma_read();
+	param_val = hw_support;
 
 	/* Fetch the value of environment variable set by the user if any. */
-	ret = fi_param_get_bool(&efa_prov, "use_device_rdma", &use_device_rdma);
+	ret = fi_param_get_bool(&efa_prov, "use_device_rdma", &param_val);
 	if (ret == -EINVAL){
 		fprintf(stderr, "FI_EFA_USE_DEVICE_RDMA was set to an invalid value by the user."
 			" FI_EFA_USE_DEVICE_RDMA is boolean and can be set to only 0/false/no/off or"
 			" 1/true/yes/on.\n");
 		abort();
 	}
+	if (ret == -FI_ENODATA) return hw_support;
```

Comment:
This is not correct. For device ID that is 0xefa0 and 0xefa1, we do not enable RDMA by default even if they support RDMA, this is to maintain consistent behavior.

Reply:
One thing need to be considered is the API version.

When user use 1.18 API and cuda is initialized, we current support FI_HMEM. In this case, we need to enable rdma if device support it.

Reply:
done.